### PR TITLE
BAEL-4448: Added examples for setting TLS version in HttpClient

### PR DESCRIPTION
**Pull request by thejoeflow**
(cherry picked from commit f4d40fc3f3140fd046ed957030e9a54582bd4a67)

### CODE REVIEW

File: httpclient/src/main/java/com/baeldung/ClientTlsVersionExamples [hunk 1]
```diff
@@ -0,0 +1,71 @@
+package com.baeldung
```
Comment (KevinGilmore) [inline httpclient/src/main/java/com/baeldung/ClientTlsVersionExamples:1]: The file should be named as a .java file, and the package should be specific to the article, e.g. com.baeldung.tlsversion

File: httpclient/src/main/java/com/baeldung/ClientTlsVersionExamples [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import java.net.SocketAddress;
+
+public class ClientTlsVersionExmaples{
+
+    public static CloseableHttpClient setViaSocketFactory(){
+        SSLConnectionSocketFactory sslsf = new SSLConnectionSocketFactory(
+                SSLContexts.createDefault(),
+                new String[] {"TLSv1.2", "TLSv1.3"},
+                null,
+                SSLConnectionSocketFactory.getDefaultHostnameVerifier());
```
Comment (KevinGilmore) [inline httpclient/src/main/java/com/baeldung/ClientTlsVersionExamples:23]: these should indent only two spaces from where the code in line 19 begins

File: httpclient/src/main/java/com/baeldung/ClientTlsVersionExamples [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+                String hostname = socket.getInetAddress().getHostName();
+                if (hostname.endsWith("internal.system.com")){
+                    socket.setEnabledProtocols(new String[] { "TLSv1", "TLSv1.1", "TLSv1.2", "TLSv1.3" });
+                }
+                else {
+                    socket.setEnabledProtocols(new String[] {"TLSv1.3"});
+                }
+            }
+        };
```
Comment (KevinGilmore) [inline httpclient/src/main/java/com/baeldung/ClientTlsVersionExamples:42]: no semicolon at end of curly brace

**Review by KevinGilmore — CHANGES_REQUESTED** [review]
FIx filename, package name, indentation etc.

File: httpclient/src/main/java/com/baeldung/ClientTlsVersionExamples [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+        CloseableHttpClient httpClient = setTlsVersionPerConnection();
+        //HttpClient httpClient = setViaSocketFactory();
+        //HttpClient httpClient = setViaSystemProperties();
+
+        try {
+            try (CloseableHttpResponse response = httpClient.execute(new HttpGet("https://httpbin.org/"))) {
+                HttpEntity entity = response.getEntity();
+                EntityUtils.consume(entity);
+            }
```
Comment (KevinGilmore) [inline httpclient/src/main/java/com/baeldung/ClientTlsVersionExamples:65]: Do you really need a nested try-with-resources inside a try?

File: httpclient/src/main/java/com/baeldung/ClientTlsVersionExamples (hunk 1 above)
Comment (thejoeflow) [reply httpclient/src/main/java/com/baeldung/ClientTlsVersionExamples]: Moved to com.baeldung.tlsversion

File: httpclient/src/main/java/com/baeldung/ClientTlsVersionExamples (hunk 2 above)
Comment (thejoeflow) [reply httpclient/src/main/java/com/baeldung/ClientTlsVersionExamples]: fixed throughout file

File: httpclient/src/main/java/com/baeldung/ClientTlsVersionExamples (hunk 4 above)
Comment (thejoeflow) [reply httpclient/src/main/java/com/baeldung/ClientTlsVersionExamples]: I did it so I could have the instantiation of the httpClient separated and I could show the two other ways of creating the client right beside it (but commented out). I changed it to a multi-line try-with-resources instead like you suggested

**Comment by KevinGilmore** [comment]
retest this please

**Review by KevinGilmore — APPROVED** [review]

**Pull request merged**